File: src/dataset/c/pangram_exercism-EzequielMarcelo/gemini/GeminiFlash_pangram_v1.c
#include "pangram.h"
#include <ctype.h>
#include <stdbool.h>

bool is_pangram(const char *sentence) {
  if (sentence == NULL) {
    return false;
  }

  int seen = 0;
  int count = 0;

  while (*sentence) {
    unsigned char c = (unsigned char)tolower(*sentence);
    if (c >= 'a' && c <= 'z') {
      int index = c - 'a';
      if (!(seen & (1 << index))) {
        seen |= (1 << index);
        count++;
        if (count == ALPHABET_COUNT) {
          return true;
        }
      }
    }
    sentence++;
  }

  return false;
}

File: src/dataset/typescript/pangram_exercism-samajammin/gemini/GeminiFlash_pangram_v1.ts
export default class Pangram {
  private readonly phrase: string;

  constructor(phrase: string) {
    this.phrase = phrase.toLowerCase();
  }

  isPangram(): boolean {
    const seen = new Array(26).fill(false);
    let count = 0;

    for (let i = 0; i < this.phrase.length; i++) {
      const charCode = this.phrase.charCodeAt(i);
      if (charCode >= 97 && charCode <= 122) {
        const index = charCode - 97;
        if (!seen[index]) {
          seen[index] = true;
          count++;
          if (count === 26) {
            return true;
          }
        }
      }
    }

    return false;
  }
}